p2pspec.md

```diff
@@ -11,23 +19,30 @@
 - A<->B ping
   - periodic message to indicate the connection is still alive
 
-:blocks:
-- A->B request block
-  - B server caches request {A, blockrange} and starts serving
-  - A should spam this request to ensure receipt
-    - can stop spamming when first block is received
-- B->A send block (either in response to request or when new block is made)
-  - how should A re-request dropped blocks?
+:: Verifier to Verifier/Player ::
+
+- send_blocks [Block]
+  - sends one or more blocks
+  - those may either be newly mined blocks, or after a request for block information
+  
+:: Verifier/Player to Verifier::
+
+- request_block_range {MyHead, MyHead - 10, UnknownBlock}
+  - requests a range of unknown blocks leading up to UnknownBlock
+  - verifier should respond with send_blocks
+  - verifier should calculate a range blocks are missing
+  - if verifier doesn't know MyHead and MyHead - 10, the response is ALL the blocks starting from height 0
 
-:potato TCP msg:
-- TCP nonsense built over UDP
-  - prob easier just to use TCP...
+:: Player to Verifier::
 
-:txs:
-- A->B send tx (that's it)
+- send_transactions [Transaction]
+  - sends a list of transactions from a player to a verifier
```

Comment:
:o

Reply:
feature creep

Reply:
it is? How would a verifier get a transactoion in the first place?

Reply:
the feature creep is that you changed it to sending a list of transactions instead of 1 at a time 🙀

Reply:
🙀

Reply:
🙀

Reply:
🙀

Reply:
🙀

Reply:
[🙀 🙀 🙀 🙀 🙀 🙀 🙀 🙀 🙀 🙀 🙀 🙀 🙀 🙀 🙀 🙀 🙀 🙀 🙀 🙀 🙀 ]

Reply:
can send a bunch of signleton messages 😱 
just didn't wanna rewrite it in case we do TCP after all and decide to bundle those

Reply:
[🙀, 🙀, 🙀, 🙀, 🙀, 🙀]

Reply:
what about
[[🙀],[🙀],[🙀],[🙀],[🙀],[🙀],[🙀],[🙀],[🙀],[🙀],[🙀],[🙀]]?